--- libint-v2.6.0-cp2k-lmax-7/configure-libint-hsw-gnu.sh	2020-05-19 17:14:10.906129533 -0700
+++ patch-files/configure-libint-hsw-gnu.sh	2020-05-19 17:15:36.693762540 -0700
@@ -12,7 +12,7 @@
 
 if [ "" = "$1" ]; then PRFX=gnu-; else PRFX=$1-; shift; fi
 HERE=$(cd $(dirname $0); pwd -P)
-DEST=${HERE}/../libint/${PRFX}hsw
+DEST=${HERE}/../libint
 
 if [[ ((! -e ${HERE}/configure.in) && (! -e ${HERE}/configure.ac)) || ("${HERE}" != "$(pwd -P)") ]]; then
   echo "Error: XCONFIGURE scripts must be located and executed in the application folder!"
@@ -28,9 +28,9 @@
 fi
 
 CONFOPTS=""
-TARGET="-march=core-avx2"
+#TARGET="-march=core-avx2"
 
-export FLAGS="-O3 ${TARGET}"
+export FLAGS="-O0 ${TARGET}"
 export LDFLAGS=""
 export CFLAGS="${FLAGS}"
 export CXXFLAGS="${FLAGS}"
