--- cp2k-2.5.1/arch/Linux-x86-64-intel.popt	2014-02-12 09:29:10.960259000 -0800
+++ patch-files/Linux-x86-64-intel.popt	2015-02-06 14:08:21.703885411 -0800
@@ -18,19 +18,18 @@
 FC       = mpif90 
 LD       = mpif90
 AR       = ar -r
+INTEL_LIB = MKL_ROOT/lib/intel64
 DFLAGS   = -D__INTEL -D__FFTSG -D__parallel -D__BLACS -D__SCALAPACK -D__FFTW3
 CPPFLAGS = 
-FCFLAGS  = $(DFLAGS) -I$(INTEL_INC) -O3 -xW -heap-arrays 64 -funroll-loops -fpp -free
-FCFLAGS2 = $(DFLAGS) -I$(INTEL_INC) -O1 -xW -heap-arrays 64 -fpp -free
-LDFLAGS  = $(FCFLAGS) -I$(INTEL_INC)
-LIBS     = $(INTEL_LIB)/libscalapack.a \
-           $(INTEL_LIB)/libblacs_init.a \
-           $(INTEL_LIB)/libblacs.a \
-           $(INTEL_LIB)/libacml.a\
-           $(INTEL_LIB)/libacml_mv.a \
-           $(INTEL_LIB)/libfftw3.a
 
 
+INTEL_LIB = MKL_ROOT/lib/intel64
+FFTW_INC = ROLLFFTW/include
+FCFLAGS  = $(DFLAGS) -I$(FFTW_INC) -O2 -xW -heap-arrays 64 -funroll-loops -fpp -free 
+FCFLAGS2 = $(DFLAGS) -I$(FFTW_INC) -O1 -xW -heap-arrays 64 -fpp -free 
+LDFLAGS  = $(FCFLAGS)
+
+LIBS     = -L $(INTEL_LIB) -lmkl_scalapack_lp64 -lmkl_rt  -lmkl_blacsROLLTYPEMKLlp64 -lpthread  -LROLLFFTW/lib -lfftw3 ROLLWRAPPER/libint_cpp_wrapper.o  -LROLLLIBINT -lderiv -lint -lstdc++
 
 graphcon.o: graphcon.F
 	$(FC) -c $(FCFLAGS2) $<
