--- cp2k-2.4.0/arch/Linux-x86-64-pgi.popt	2007-10-25 01:42:00.000000000 -0700
+++ Linux-x86-64-pgi.popt	2014-02-01 13:56:16.000000000 -0800
@@ -6,14 +6,14 @@
 #
 CC       = cc
 CPP      = cpp
-FC       = pgf90 -Mfree
-LD       = pgf90
+FC       = mpif90 -Mfree
+LD       = mpif90
 AR       = ar -r
 DFLAGS   = -D__PGI -D__FFTSG\
            -D__parallel -D__BLACS -D__SCALAPACK\
-           -D__FFTW3
+           -D__FFTW3 -D__LIBINT
 CPPFLAGS = -traditional -C $(DFLAGS) -P
-FCFLAGS  = -fastsse -tp amd64
+FCFLAGS  = -fastsse -tp amd64 -I$(FFTW_INC) -O2 -heap-arrays 64 -funroll-loops -fpp -free -I../libint/include -IROLLROOT/src/cp2k/cp2k.VERSION/obj/Linux-x86-64-ROLLCOMPILER/popt
 LDFLAGS  = $(FCFLAGS) -Mmpi -Mscalapack
 LIBS     = -lacml -ldfftw
 
