--- lammps-16Feb16/lib/gpu/Makefile.linux	2014-01-28 16:07:38.000000000 -0800
+++ patch-files/gpu/Makefile.linux	2016-06-11 19:30:54.359580735 -0700
@@ -7,11 +7,12 @@
 
 EXTRAMAKE = Makefile.lammps.standard
 
-CUDA_HOME = /usr/local/cuda
+CUDA_HOME = ${CUDAHOME}
 NVCC = nvcc
 
+CUDA_ARCH = -gencode arch=compute_30,code=sm_30
 # Tesla CUDA
-CUDA_ARCH = -arch=sm_21
+#CUDA_ARCH = -arch=sm_21
 # newer CUDA
 #CUDA_ARCH = -arch=sm_13
 # older CUDA
@@ -33,7 +34,7 @@
 CUDA_LIB = -L$(CUDA_HOME)/lib64
 CUDA_OPTS = -DUNIX -O3 -Xptxas -v --use_fast_math $(LMP_INC)
 
-CUDR_CPP = mpic++ -DMPI_GERYON -DUCL_NO_EXIT -DMPICH_IGNORE_CXX_SEEK
+CUDR_CPP = mpicxx -DMPI_GERYON -DUCL_NO_EXIT -DMPICH_IGNORE_CXX_SEEK
 CUDR_OPTS = -O2 $(LMP_INC) # -xHost -no-prec-div -ansi-alias
 
 BIN_DIR = ./
