--- lammps-31Mar17/lib/gpu/Makefile.linux	2016-07-01 16:27:26.000000000 -0700
+++ patch-files/gpu/Makefile.linux	2017-07-11 13:32:37.240988450 -0700
@@ -7,19 +7,12 @@
 
 EXTRAMAKE = Makefile.lammps.standard
 
-ifeq ($(CUDA_HOME),)
-CUDA_HOME = /usr/local/cuda
-endif
+CUDA_HOME = ${CUDAHOME}
 
 NVCC = nvcc
 
 # Tesla CUDA
-CUDA_ARCH = -arch=sm_21
-# newer CUDA
-#CUDA_ARCH = -arch=sm_13
-# older CUDA
-#CUDA_ARCH = -arch=sm_10 -DCUDA_PRE_THREE
-CUDA_ARCH = -arch=sm_35
+CUDA_ARCH = -gencode arch=compute_CUDA_CAPABILITY,code=sm_CUDA_CAPABILITY
 
 # this setting should match LAMMPS Makefile
 # one of LAMMPS_SMALLBIG (default), LAMMPS_BIGBIG and LAMMPS_SMALLSMALL
@@ -34,10 +27,10 @@
 CUDA_PRECISION = -D_SINGLE_DOUBLE
 
 CUDA_INCLUDE = -I$(CUDA_HOME)/include
-CUDA_LIB = -L$(CUDA_HOME)/lib64
+CUDA_LIB = -L$(CUDA_HOME)/lib64 -L/usr/lib64/nvidia -lnvidia-fatbinaryloader
 CUDA_OPTS = -DUNIX -O3 -Xptxas -v --use_fast_math $(LMP_INC)
 
-CUDR_CPP = mpic++ -DMPI_GERYON -DUCL_NO_EXIT -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1 -fPIC
+CUDR_CPP = mpicxx -DMPI_GERYON -DUCL_NO_EXIT -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1 -fPIC
 CUDR_OPTS = -O2 $(LMP_INC) # -xHost -no-prec-div -ansi-alias
 
 BIN_DIR = ./
