--- lammps-22Aug18/lib/gpu/Makefile.linux	2018-08-15 14:26:05.000000000 -0700
+++ patch-files/gpu/Makefile.linux	2018-08-28 17:20:50.494007586 -0700
@@ -13,13 +13,7 @@
 
 NVCC = nvcc
 
-# older CUDA
-#CUDA_ARCH = -arch=sm_21
-# newer CUDA
-#CUDA_ARCH = -arch=sm_13
-# older CUDA
-#CUDA_ARCH = -arch=sm_10 -DCUDA_PRE_THREE
-CUDA_ARCH = -arch=sm_35
+CUDA_ARCH = -arch=sm_CUDA_CAPABILITY
 
 # this setting should match LAMMPS Makefile
 # one of LAMMPS_SMALLBIG (default), LAMMPS_BIGBIG and LAMMPS_SMALLSMALL
@@ -34,7 +28,7 @@
 CUDA_PRECISION = -D_SINGLE_DOUBLE
 
 CUDA_INCLUDE = -I$(CUDA_HOME)/include
-CUDA_LIB = -L$(CUDA_HOME)/lib64
+CUDA_LIB = -L$(CUDA_HOME)/lib64 EXTRACUDAPATH EXTRACUDALIBS
 CUDA_OPTS = -DUNIX -O3 -Xptxas -v --use_fast_math $(LMP_INC) -Xcompiler -fPIC
 
 CUDR_CPP = mpicxx -DMPI_GERYON -DUCL_NO_EXIT -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1 -fPIC
