--- lammps-31Mar17/lib/gpu/Nvidia.makefile	2016-09-08 13:20:32.000000000 -0700
+++ patch-files/gpu/Nvidia.makefile	2017-07-11 10:34:23.477673195 -0700
@@ -155,7 +155,7 @@
 	$(CUDA) -o $@ -c cudpp_mini/scan_app.cu
 
 $(OBJ_DIR)/atom.cubin: lal_atom.cu lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_atom.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_atom.cu
 
 $(OBJ_DIR)/atom_cubin.h: $(OBJ_DIR)/atom.cubin
 	$(BIN2C) -c -n atom $(OBJ_DIR)/atom.cubin > $(OBJ_DIR)/atom_cubin.h
@@ -167,13 +167,13 @@
 	$(CUDR) -o $@ -c lal_answer.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/neighbor_cpu.cubin: lal_neighbor_cpu.cu lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_neighbor_cpu.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_neighbor_cpu.cu
 
 $(OBJ_DIR)/neighbor_cpu_cubin.h: $(OBJ_DIR)/neighbor_cpu.cubin
 	$(BIN2C) -c -n neighbor_cpu $(OBJ_DIR)/neighbor_cpu.cubin > $(OBJ_DIR)/neighbor_cpu_cubin.h
 
 $(OBJ_DIR)/neighbor_gpu.cubin: lal_neighbor_gpu.cu lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_neighbor_gpu.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_neighbor_gpu.cu
 
 $(OBJ_DIR)/neighbor_gpu_cubin.h: $(OBJ_DIR)/neighbor_gpu.cubin
 	$(BIN2C) -c -n neighbor_gpu $(OBJ_DIR)/neighbor_gpu.cubin > $(OBJ_DIR)/neighbor_gpu_cubin.h
@@ -185,7 +185,7 @@
 	$(CUDR) -o $@ -c lal_neighbor.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/device.cubin: lal_device.cu lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_device.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_device.cu
 
 $(OBJ_DIR)/device_cubin.h: $(OBJ_DIR)/device.cubin
 	$(BIN2C) -c -n device $(OBJ_DIR)/device.cubin > $(OBJ_DIR)/device_cubin.h
@@ -212,13 +212,13 @@
 	$(CUDR) -o $@ -c lal_base_dpd.cpp
 
 $(OBJ_DIR)/pppm_f.cubin: lal_pppm.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -Dgrdtyp=float -Dgrdtyp4=float4 -o $@ lal_pppm.cu
+	$(CUDA)  -DNV_KERNEL -Dgrdtyp=float -Dgrdtyp4=float4 -o $@ lal_pppm.cu
 
 $(OBJ_DIR)/pppm_f_cubin.h: $(OBJ_DIR)/pppm_f.cubin
 	$(BIN2C) -c -n pppm_f $(OBJ_DIR)/pppm_f.cubin > $(OBJ_DIR)/pppm_f_cubin.h
 
 $(OBJ_DIR)/pppm_d.cubin: lal_pppm.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -Dgrdtyp=double -Dgrdtyp4=double4 -o $@ lal_pppm.cu
+	$(CUDA)  -DNV_KERNEL -Dgrdtyp=double -Dgrdtyp4=double4 -o $@ lal_pppm.cu
 
 $(OBJ_DIR)/pppm_d_cubin.h: $(OBJ_DIR)/pppm_d.cubin
 	$(BIN2C) -c -n pppm_d $(OBJ_DIR)/pppm_d.cubin > $(OBJ_DIR)/pppm_d_cubin.h
@@ -230,16 +230,16 @@
 	$(CUDR) -o $@ -c lal_pppm_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/ellipsoid_nbor.cubin: lal_ellipsoid_nbor.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_ellipsoid_nbor.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_ellipsoid_nbor.cu
 
 $(OBJ_DIR)/ellipsoid_nbor_cubin.h: $(OBJ_DIR)/ellipsoid_nbor.cubin
 	$(BIN2C) -c -n ellipsoid_nbor $(OBJ_DIR)/ellipsoid_nbor.cubin > $(OBJ_DIR)/ellipsoid_nbor_cubin.h
 
 $(OBJ_DIR)/gayberne.cubin: lal_gayberne.cu lal_precision.h lal_ellipsoid_extra.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_gayberne.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_gayberne.cu
 
 $(OBJ_DIR)/gayberne_lj.cubin: lal_gayberne_lj.cu lal_precision.h lal_ellipsoid_extra.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_gayberne_lj.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_gayberne_lj.cu
 
 $(OBJ_DIR)/gayberne_cubin.h: $(OBJ_DIR)/gayberne.cubin
 	$(BIN2C) -c -n gayberne $(OBJ_DIR)/gayberne.cubin > $(OBJ_DIR)/gayberne_cubin.h
@@ -254,10 +254,10 @@
 	$(CUDR) -o $@ -c lal_gayberne_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/re_squared.cubin: lal_re_squared.cu lal_precision.h lal_ellipsoid_extra.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_re_squared.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_re_squared.cu
 
 $(OBJ_DIR)/re_squared_lj.cubin: lal_re_squared_lj.cu lal_precision.h lal_ellipsoid_extra.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_re_squared_lj.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_re_squared_lj.cu
 
 $(OBJ_DIR)/re_squared_cubin.h: $(OBJ_DIR)/re_squared.cubin
 	$(BIN2C) -c -n re_squared $(OBJ_DIR)/re_squared.cubin > $(OBJ_DIR)/re_squared_cubin.h
@@ -272,7 +272,7 @@
 	$(CUDR) -o $@ -c lal_re_squared_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj.cubin: lal_lj.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj.cu
 
 $(OBJ_DIR)/lj_cubin.h: $(OBJ_DIR)/lj.cubin $(OBJ_DIR)/lj.cubin
 	$(BIN2C) -c -n lj $(OBJ_DIR)/lj.cubin > $(OBJ_DIR)/lj_cubin.h
@@ -284,7 +284,7 @@
 	$(CUDR) -o $@ -c lal_lj_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_coul.cubin: lal_lj_coul.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_coul.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_coul.cu
 
 $(OBJ_DIR)/lj_coul_cubin.h: $(OBJ_DIR)/lj_coul.cubin $(OBJ_DIR)/lj_coul.cubin
 	$(BIN2C) -c -n lj_coul $(OBJ_DIR)/lj_coul.cubin > $(OBJ_DIR)/lj_coul_cubin.h
@@ -296,7 +296,7 @@
 	$(CUDR) -o $@ -c lal_lj_coul_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_class2_long.cubin: lal_lj_class2_long.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_class2_long.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_class2_long.cu
 
 $(OBJ_DIR)/lj_class2_long_cubin.h: $(OBJ_DIR)/lj_class2_long.cubin $(OBJ_DIR)/lj_class2_long.cubin
 	$(BIN2C) -c -n lj_class2_long $(OBJ_DIR)/lj_class2_long.cubin > $(OBJ_DIR)/lj_class2_long_cubin.h
@@ -308,7 +308,7 @@
 	$(CUDR) -o $@ -c lal_lj_class2_long_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/coul_long.cubin: lal_coul_long.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_coul_long.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_coul_long.cu
 
 $(OBJ_DIR)/coul_long_cubin.h: $(OBJ_DIR)/coul_long.cubin $(OBJ_DIR)/coul_long.cubin
 	$(BIN2C) -c -n coul_long $(OBJ_DIR)/coul_long.cubin > $(OBJ_DIR)/coul_long_cubin.h
@@ -320,7 +320,7 @@
 	$(CUDR) -o $@ -c lal_coul_long_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_coul_long.cubin: lal_lj_coul_long.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_coul_long.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_coul_long.cu
 
 $(OBJ_DIR)/lj_coul_long_cubin.h: $(OBJ_DIR)/lj_coul_long.cubin $(OBJ_DIR)/lj_coul_long.cubin
 	$(BIN2C) -c -n lj_coul_long $(OBJ_DIR)/lj_coul_long.cubin > $(OBJ_DIR)/lj_coul_long_cubin.h
@@ -332,7 +332,7 @@
 	$(CUDR) -o $@ -c lal_lj_coul_long_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_dsf.cubin: lal_lj_dsf.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_dsf.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_dsf.cu
 
 $(OBJ_DIR)/lj_dsf_cubin.h: $(OBJ_DIR)/lj_dsf.cubin $(OBJ_DIR)/lj_dsf.cubin
 	$(BIN2C) -c -n lj_dsf $(OBJ_DIR)/lj_dsf.cubin > $(OBJ_DIR)/lj_dsf_cubin.h
@@ -344,7 +344,7 @@
 	$(CUDR) -o $@ -c lal_lj_dsf_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/morse.cubin: lal_morse.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_morse.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_morse.cu
 
 $(OBJ_DIR)/morse_cubin.h: $(OBJ_DIR)/morse.cubin $(OBJ_DIR)/morse.cubin
 	$(BIN2C) -c -n morse $(OBJ_DIR)/morse.cubin > $(OBJ_DIR)/morse_cubin.h
@@ -356,7 +356,7 @@
 	$(CUDR) -o $@ -c lal_morse_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/charmm_long.cubin: lal_charmm_long.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_charmm_long.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_charmm_long.cu
 
 $(OBJ_DIR)/charmm_long_cubin.h: $(OBJ_DIR)/charmm_long.cubin $(OBJ_DIR)/charmm_long.cubin
 	$(BIN2C) -c -n charmm_long $(OBJ_DIR)/charmm_long.cubin > $(OBJ_DIR)/charmm_long_cubin.h
@@ -368,7 +368,7 @@
 	$(CUDR) -o $@ -c lal_charmm_long_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj96.cubin: lal_lj96.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj96.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj96.cu
 
 $(OBJ_DIR)/lj96_cubin.h: $(OBJ_DIR)/lj96.cubin $(OBJ_DIR)/lj96.cubin
 	$(BIN2C) -c -n lj96 $(OBJ_DIR)/lj96.cubin > $(OBJ_DIR)/lj96_cubin.h
@@ -380,7 +380,7 @@
 	$(CUDR) -o $@ -c lal_lj96_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_expand.cubin: lal_lj_expand.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_expand.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_expand.cu
 
 $(OBJ_DIR)/lj_expand_cubin.h: $(OBJ_DIR)/lj_expand.cubin $(OBJ_DIR)/lj_expand.cubin
 	$(BIN2C) -c -n lj_expand $(OBJ_DIR)/lj_expand.cubin > $(OBJ_DIR)/lj_expand_cubin.h
@@ -392,7 +392,7 @@
 	$(CUDR) -o $@ -c lal_lj_expand_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/cg_cmm.cubin: lal_cg_cmm.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_cg_cmm.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_cg_cmm.cu
 
 $(OBJ_DIR)/cg_cmm_cubin.h: $(OBJ_DIR)/cg_cmm.cubin $(OBJ_DIR)/cg_cmm.cubin
 	$(BIN2C) -c -n cg_cmm $(OBJ_DIR)/cg_cmm.cubin > $(OBJ_DIR)/cg_cmm_cubin.h
@@ -404,7 +404,7 @@
 	$(CUDR) -o $@ -c lal_cg_cmm_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/cg_cmm_long.cubin: lal_cg_cmm_long.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_cg_cmm_long.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_cg_cmm_long.cu
 
 $(OBJ_DIR)/cg_cmm_long_cubin.h: $(OBJ_DIR)/cg_cmm_long.cubin $(OBJ_DIR)/cg_cmm_long.cubin
 	$(BIN2C) -c -n cg_cmm_long $(OBJ_DIR)/cg_cmm_long.cubin > $(OBJ_DIR)/cg_cmm_long_cubin.h
@@ -416,7 +416,7 @@
 	$(CUDR) -o $@ -c lal_cg_cmm_long_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/eam.cubin: lal_eam.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_eam.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_eam.cu
 
 $(OBJ_DIR)/eam_cubin.h: $(OBJ_DIR)/eam.cubin $(OBJ_DIR)/eam.cubin
 	$(BIN2C) -c -n eam $(OBJ_DIR)/eam.cubin > $(OBJ_DIR)/eam_cubin.h
@@ -434,7 +434,7 @@
 	$(CUDR) -o $@ -c lal_eam_alloy_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/buck.cubin: lal_buck.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_buck.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_buck.cu
 
 $(OBJ_DIR)/buck_cubin.h: $(OBJ_DIR)/buck.cubin $(OBJ_DIR)/buck.cubin
 	$(BIN2C) -c -n buck $(OBJ_DIR)/buck.cubin > $(OBJ_DIR)/buck_cubin.h
@@ -446,7 +446,7 @@
 	$(CUDR) -o $@ -c lal_buck_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/buck_coul.cubin: lal_buck_coul.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_buck_coul.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_buck_coul.cu
 
 $(OBJ_DIR)/buck_coul_cubin.h: $(OBJ_DIR)/buck_coul.cubin $(OBJ_DIR)/buck_coul.cubin
 	$(BIN2C) -c -n buck_coul $(OBJ_DIR)/buck_coul.cubin > $(OBJ_DIR)/buck_coul_cubin.h
@@ -458,7 +458,7 @@
 	$(CUDR) -o $@ -c lal_buck_coul_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/buck_coul_long.cubin: lal_buck_coul_long.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_buck_coul_long.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_buck_coul_long.cu
 
 $(OBJ_DIR)/buck_coul_long_cubin.h: $(OBJ_DIR)/buck_coul_long.cubin $(OBJ_DIR)/buck_coul_long.cubin
 	$(BIN2C) -c -n buck_coul_long $(OBJ_DIR)/buck_coul_long.cubin > $(OBJ_DIR)/buck_coul_long_cubin.h
@@ -470,7 +470,7 @@
 	$(CUDR) -o $@ -c lal_buck_coul_long_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/table.cubin: lal_table.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_table.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_table.cu
 
 $(OBJ_DIR)/table_cubin.h: $(OBJ_DIR)/table.cubin $(OBJ_DIR)/table.cubin
 	$(BIN2C) -c -n table $(OBJ_DIR)/table.cubin > $(OBJ_DIR)/table_cubin.h
@@ -482,7 +482,7 @@
 	$(CUDR) -o $@ -c lal_table_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/yukawa.cubin: lal_yukawa.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_yukawa.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_yukawa.cu
 
 $(OBJ_DIR)/yukawa_cubin.h: $(OBJ_DIR)/yukawa.cubin $(OBJ_DIR)/yukawa.cubin
 	$(BIN2C) -c -n yukawa $(OBJ_DIR)/yukawa.cubin > $(OBJ_DIR)/yukawa_cubin.h
@@ -494,7 +494,7 @@
 	$(CUDR) -o $@ -c lal_yukawa_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/born.cubin: lal_born.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_born.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_born.cu
 
 $(OBJ_DIR)/born_cubin.h: $(OBJ_DIR)/born.cubin $(OBJ_DIR)/born.cubin
 	$(BIN2C) -c -n born $(OBJ_DIR)/born.cubin > $(OBJ_DIR)/born_cubin.h
@@ -506,7 +506,7 @@
 	$(CUDR) -o $@ -c lal_born_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/born_coul_wolf.cubin: lal_born_coul_wolf.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_born_coul_wolf.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_born_coul_wolf.cu
 
 $(OBJ_DIR)/born_coul_wolf_cubin.h: $(OBJ_DIR)/born_coul_wolf.cubin $(OBJ_DIR)/born_coul_wolf.cubin
 	$(BIN2C) -c -n born_coul_wolf $(OBJ_DIR)/born_coul_wolf.cubin > $(OBJ_DIR)/born_coul_wolf_cubin.h
@@ -518,7 +518,7 @@
 	$(CUDR) -o $@ -c lal_born_coul_wolf_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/born_coul_long.cubin: lal_born_coul_long.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_born_coul_long.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_born_coul_long.cu
 
 $(OBJ_DIR)/born_coul_long_cubin.h: $(OBJ_DIR)/born_coul_long.cubin $(OBJ_DIR)/born_coul_long.cubin
 	$(BIN2C) -c -n born_coul_long $(OBJ_DIR)/born_coul_long.cubin > $(OBJ_DIR)/born_coul_long_cubin.h
@@ -530,7 +530,7 @@
 	$(CUDR) -o $@ -c lal_born_coul_long_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/dipole_lj.cubin: lal_dipole_lj.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_dipole_lj.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_dipole_lj.cu
 
 $(OBJ_DIR)/dipole_lj_cubin.h: $(OBJ_DIR)/dipole_lj.cubin $(OBJ_DIR)/dipole_lj.cubin
 	$(BIN2C) -c -n dipole_lj $(OBJ_DIR)/dipole_lj.cubin > $(OBJ_DIR)/dipole_lj_cubin.h
@@ -542,7 +542,7 @@
 	$(CUDR) -o $@ -c lal_dipole_lj_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/dipole_lj_sf.cubin: lal_dipole_lj_sf.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_dipole_lj_sf.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_dipole_lj_sf.cu
 
 $(OBJ_DIR)/dipole_lj_sf_cubin.h: $(OBJ_DIR)/dipole_lj_sf.cubin $(OBJ_DIR)/dipole_lj_sf.cubin
 	$(BIN2C) -c -n dipole_lj_sf $(OBJ_DIR)/dipole_lj_sf.cubin > $(OBJ_DIR)/dipole_lj_sf_cubin.h
@@ -554,7 +554,7 @@
 	$(CUDR) -o $@ -c lal_dipole_lj_sf_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/colloid.cubin: lal_colloid.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_colloid.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_colloid.cu
 
 $(OBJ_DIR)/colloid_cubin.h: $(OBJ_DIR)/colloid.cubin $(OBJ_DIR)/colloid.cubin
 	$(BIN2C) -c -n colloid $(OBJ_DIR)/colloid.cubin > $(OBJ_DIR)/colloid_cubin.h
@@ -566,7 +566,7 @@
 	$(CUDR) -o $@ -c lal_colloid_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/gauss.cubin: lal_gauss.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_gauss.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_gauss.cu
 
 $(OBJ_DIR)/gauss_cubin.h: $(OBJ_DIR)/gauss.cubin $(OBJ_DIR)/gauss.cubin
 	$(BIN2C) -c -n gauss $(OBJ_DIR)/gauss.cubin > $(OBJ_DIR)/gauss_cubin.h
@@ -578,7 +578,7 @@
 	$(CUDR) -o $@ -c lal_gauss_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/yukawa_colloid.cubin: lal_yukawa_colloid.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_yukawa_colloid.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_yukawa_colloid.cu
 
 $(OBJ_DIR)/yukawa_colloid_cubin.h: $(OBJ_DIR)/yukawa_colloid.cubin $(OBJ_DIR)/yukawa_colloid.cubin
 	$(BIN2C) -c -n yukawa_colloid $(OBJ_DIR)/yukawa_colloid.cubin > $(OBJ_DIR)/yukawa_colloid_cubin.h
@@ -590,7 +590,7 @@
 	$(CUDR) -o $@ -c lal_yukawa_colloid_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_coul_debye.cubin: lal_lj_coul_debye.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_coul_debye.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_coul_debye.cu
 
 $(OBJ_DIR)/lj_coul_debye_cubin.h: $(OBJ_DIR)/lj_coul_debye.cubin $(OBJ_DIR)/lj_coul_debye.cubin
 	$(BIN2C) -c -n lj_coul_debye $(OBJ_DIR)/lj_coul_debye.cubin > $(OBJ_DIR)/lj_coul_debye_cubin.h
@@ -602,7 +602,7 @@
 	$(CUDR) -o $@ -c lal_lj_coul_debye_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/coul_dsf.cubin: lal_coul_dsf.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_coul_dsf.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_coul_dsf.cu
 
 $(OBJ_DIR)/coul_dsf_cubin.h: $(OBJ_DIR)/coul_dsf.cubin $(OBJ_DIR)/coul_dsf.cubin
 	$(BIN2C) -c -n coul_dsf $(OBJ_DIR)/coul_dsf.cubin > $(OBJ_DIR)/coul_dsf_cubin.h
@@ -614,7 +614,7 @@
 	$(CUDR) -o $@ -c lal_coul_dsf_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/sw.cubin: lal_sw.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_sw.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_sw.cu
 
 $(OBJ_DIR)/sw_cubin.h: $(OBJ_DIR)/sw.cubin $(OBJ_DIR)/sw.cubin
 	$(BIN2C) -c -n sw $(OBJ_DIR)/sw.cubin > $(OBJ_DIR)/sw_cubin.h
@@ -626,7 +626,7 @@
 	$(CUDR) -o $@ -c lal_sw_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/beck.cubin: lal_beck.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_beck.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_beck.cu
 
 $(OBJ_DIR)/beck_cubin.h: $(OBJ_DIR)/beck.cubin $(OBJ_DIR)/beck.cubin
 	$(BIN2C) -c -n beck $(OBJ_DIR)/beck.cubin > $(OBJ_DIR)/beck_cubin.h
@@ -638,7 +638,7 @@
 	$(CUDR) -o $@ -c lal_beck_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/mie.cubin: lal_mie.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_mie.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_mie.cu
 
 $(OBJ_DIR)/mie_cubin.h: $(OBJ_DIR)/mie.cubin $(OBJ_DIR)/mie.cubin
 	$(BIN2C) -c -n mie $(OBJ_DIR)/mie.cubin > $(OBJ_DIR)/mie_cubin.h
@@ -650,7 +650,7 @@
 	$(CUDR) -o $@ -c lal_mie_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/soft.cubin: lal_soft.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_soft.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_soft.cu
 
 $(OBJ_DIR)/soft_cubin.h: $(OBJ_DIR)/soft.cubin $(OBJ_DIR)/soft.cubin
 	$(BIN2C) -c -n soft $(OBJ_DIR)/soft.cubin > $(OBJ_DIR)/soft_cubin.h
@@ -662,7 +662,7 @@
 	$(CUDR) -o $@ -c lal_soft_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_coul_msm.cubin: lal_lj_coul_msm.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_coul_msm.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_coul_msm.cu
 
 $(OBJ_DIR)/lj_coul_msm_cubin.h: $(OBJ_DIR)/lj_coul_msm.cubin $(OBJ_DIR)/lj_coul_msm.cubin
 	$(BIN2C) -c -n lj_coul_msm $(OBJ_DIR)/lj_coul_msm.cubin > $(OBJ_DIR)/lj_coul_msm_cubin.h
@@ -674,7 +674,7 @@
 	$(CUDR) -o $@ -c lal_lj_coul_msm_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_gromacs.cubin: lal_lj_gromacs.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_gromacs.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_gromacs.cu
 
 $(OBJ_DIR)/lj_gromacs_cubin.h: $(OBJ_DIR)/lj_gromacs.cubin $(OBJ_DIR)/lj_gromacs.cubin
 	$(BIN2C) -c -n lj_gromacs $(OBJ_DIR)/lj_gromacs.cubin > $(OBJ_DIR)/lj_gromacs_cubin.h
@@ -686,7 +686,7 @@
 	$(CUDR) -o $@ -c lal_lj_gromacs_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/dpd.cubin: lal_dpd.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_dpd.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_dpd.cu
 
 $(OBJ_DIR)/dpd_cubin.h: $(OBJ_DIR)/dpd.cubin $(OBJ_DIR)/dpd.cubin
 	$(BIN2C) -c -n dpd $(OBJ_DIR)/dpd.cubin > $(OBJ_DIR)/dpd_cubin.h
@@ -698,7 +698,7 @@
 	$(CUDR) -o $@ -c lal_dpd_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/tersoff.cubin: lal_tersoff.cu lal_precision.h lal_tersoff_extra.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_tersoff.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_tersoff.cu
 
 $(OBJ_DIR)/tersoff_cubin.h: $(OBJ_DIR)/tersoff.cubin $(OBJ_DIR)/tersoff.cubin
 	$(BIN2C) -c -n tersoff $(OBJ_DIR)/tersoff.cubin > $(OBJ_DIR)/tersoff_cubin.h
@@ -710,7 +710,7 @@
 	$(CUDR) -o $@ -c lal_tersoff_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/tersoff_zbl.cubin: lal_tersoff_zbl.cu lal_precision.h lal_tersoff_zbl_extra.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_tersoff_zbl.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_tersoff_zbl.cu
 
 $(OBJ_DIR)/tersoff_zbl_cubin.h: $(OBJ_DIR)/tersoff_zbl.cubin $(OBJ_DIR)/tersoff_zbl.cubin
 	$(BIN2C) -c -n tersoff_zbl $(OBJ_DIR)/tersoff_zbl.cubin > $(OBJ_DIR)/tersoff_zbl_cubin.h
@@ -722,7 +722,7 @@
 	$(CUDR) -o $@ -c lal_tersoff_zbl_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/tersoff_mod.cubin: lal_tersoff_mod.cu lal_precision.h lal_tersoff_mod_extra.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_tersoff_mod.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_tersoff_mod.cu
 
 $(OBJ_DIR)/tersoff_mod_cubin.h: $(OBJ_DIR)/tersoff_mod.cubin $(OBJ_DIR)/tersoff_mod.cubin
 	$(BIN2C) -c -n tersoff_mod $(OBJ_DIR)/tersoff_mod.cubin > $(OBJ_DIR)/tersoff_mod_cubin.h
@@ -734,7 +734,7 @@
 	$(CUDR) -o $@ -c lal_tersoff_mod_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/coul.cubin: lal_coul.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_coul.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_coul.cu
 
 $(OBJ_DIR)/coul_cubin.h: $(OBJ_DIR)/coul.cubin $(OBJ_DIR)/coul.cubin
 	$(BIN2C) -c -n coul $(OBJ_DIR)/coul.cubin > $(OBJ_DIR)/coul_cubin.h
@@ -746,7 +746,7 @@
 	$(CUDR) -o $@ -c lal_coul_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/coul_debye.cubin: lal_coul_debye.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_coul_debye.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_coul_debye.cu
 
 $(OBJ_DIR)/coul_debye_cubin.h: $(OBJ_DIR)/coul_debye.cubin $(OBJ_DIR)/coul_debye.cubin
 	$(BIN2C) -c -n coul_debye $(OBJ_DIR)/coul_debye.cubin > $(OBJ_DIR)/coul_debye_cubin.h
@@ -758,7 +758,7 @@
 	$(CUDR) -o $@ -c lal_coul_debye_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/zbl.cubin: lal_zbl.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_zbl.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_zbl.cu
 
 $(OBJ_DIR)/zbl_cubin.h: $(OBJ_DIR)/zbl.cubin $(OBJ_DIR)/zbl.cubin
 	$(BIN2C) -c -n zbl $(OBJ_DIR)/zbl.cubin > $(OBJ_DIR)/zbl_cubin.h
@@ -770,7 +770,7 @@
 	$(CUDR) -o $@ -c lal_zbl_ext.cpp -I$(OBJ_DIR)
 
 $(OBJ_DIR)/lj_cubic.cubin: lal_lj_cubic.cu lal_precision.h lal_preprocessor.h
-	$(CUDA) --cubin -DNV_KERNEL -o $@ lal_lj_cubic.cu
+	$(CUDA)  -DNV_KERNEL -o $@ lal_lj_cubic.cu
 
 $(OBJ_DIR)/lj_cubic_cubin.h: $(OBJ_DIR)/lj_cubic.cubin $(OBJ_DIR)/lj_cubic.cubin
 	$(BIN2C) -c -n lj_cubic $(OBJ_DIR)/lj_cubic.cubin > $(OBJ_DIR)/lj_cubic_cubin.h
