--- lammps-9Dec14/src/MAKE/Makefile.mpi	2014-10-06 13:59:58.000000000 -0700
+++ patch-files/Makefile.mpi	2015-02-18 15:21:19.000000000 -0800
@@ -7,13 +7,13 @@
 # specify flags and libraries needed for your compiler
 
 CC =		mpicxx
-CCFLAGS =	-g -O3 
+CCFLAGS =	-O2 -restrict
 SHFLAGS =	-fPIC
 DEPFLAGS =	-M
 
 LINK =		mpicxx
-LINKFLAGS =	-g -O
-LIB = 
+LINKFLAGS =	-O2
+LIB = 		-lstdc++ -LMKLLIB -lmkl_gf_ilp64 -lmkl_gnu_thread -lmkl_core -lpthread -lgomp
 SIZE =		size
 
 ARCHIVE =	ar
@@ -50,7 +50,7 @@
 # PATH = path for FFT library
 # LIB = name of FFT library
 
-FFT_INC =    	
+FFT_INC =    	-DFFT_MKL -IMKL_ROOT
 FFT_PATH = 
 FFT_LIB =	
 
