--- lammps-1Aug14/src/MAKE/Makefile.linux	2013-10-30 08:23:31.000000000 -0700
+++ patch-files/Makefile.linux	2014-09-18 09:30:41.000000000 -0700
@@ -6,14 +6,14 @@
 # compiler/linker settings
 # specify flags and libraries needed for your compiler
 
-CC =		icc
-CCFLAGS =	-O
+CC =		mpicxx
+CCFLAGS =	-O2
 SHFLAGS =	-fPIC
 DEPFLAGS =	-M
 
-LINK =		icc
-LINKFLAGS =	-O
-LIB =           -lstdc++
+LINK =		mpicxx
+LINKFLAGS =	-O2
+LIB =           LINK_LIBS
 SIZE =		size
 
 ARCHIVE =	ar
@@ -37,9 +37,9 @@
 # PATH = path for MPI library
 # LIB = name of MPI library
 
-MPI_INC =       -DMPICH_SKIP_MPICXX 
+MPI_INC =       
 MPI_PATH = 
-MPI_LIB =	-lmpich -lmpl -lpthread
+MPI_LIB =
 
 # FFT library, OPTIONAL
 # see discussion in doc/Section_start.html#2_2 (step 6)
@@ -48,9 +48,9 @@
 # PATH = path for FFT library
 # LIB = name of FFT library
 
-FFT_INC =       -DFFT_FFTW
+FFT_INC =       -DFFT_MKL -IMKL_HOME/include 
 FFT_PATH = 
-FFT_LIB =	-lfftw
+FFT_LIB =	
 
 # JPEG and/or PNG library, OPTIONAL
 # see discussion in doc/Section_start.html#2_2 (step 7)
