--- lammps-9Dec14/lib/gpu/Makefile.linux	2014-01-28 16:07:38.000000000 -0800
+++ patch-files/Makefile.gpu	2015-02-06 21:59:39.769855201 -0800
@@ -7,7 +7,7 @@
 
 EXTRAMAKE = Makefile.lammps.standard
 
-CUDA_HOME = /usr/local/cuda
+CUDA_HOME = ROLLCUDA
 NVCC = nvcc
 
 # Tesla CUDA
@@ -16,6 +16,7 @@
 #CUDA_ARCH = -arch=sm_13
 # older CUDA
 #CUDA_ARCH = -arch=sm_10 -DCUDA_PRE_THREE
+CUDA_ARCH = -arch=sm_30
 
 # this setting should match LAMMPS Makefile
 # one of LAMMPS_SMALLBIG (default), LAMMPS_BIGBIG and LAMMPS_SMALLSMALL
