--- lammps-31Jan14/lib/meam/Makefile.lammps.ifort	2011-08-18 10:32:34.000000000 -0700
+++ patch-files/Makefile.lammps.meam	2014-03-12 10:08:04.000000000 -0700
@@ -1,5 +1,5 @@
 # Settings that the LAMMPS build will import when this package library is used
 
 meam_SYSINC =
-meam_SYSLIB = -lifcore -lsvml -lompstub -limf
+meam_SYSLIB = -lifcore -lsvml -limf
 meam_SYSPATH = -L/opt/intel/fce/10.0.023/lib
