--- lammps-7Dec15/lib/gpu/Makefile.linux	2014-01-28 16:07:38.000000000 -0800
+++ patch-files/Makefile.linux	2016-01-07 17:06:37.053277284 -0800
@@ -7,7 +7,7 @@
 
 EXTRAMAKE = Makefile.lammps.standard
 
-CUDA_HOME = /usr/local/cuda
+CUDA_HOME = ${CUDAHOME}
 NVCC = nvcc
 
 # Tesla CUDA
@@ -33,7 +33,7 @@
 CUDA_LIB = -L$(CUDA_HOME)/lib64
 CUDA_OPTS = -DUNIX -O3 -Xptxas -v --use_fast_math $(LMP_INC)
 
-CUDR_CPP = mpic++ -DMPI_GERYON -DUCL_NO_EXIT -DMPICH_IGNORE_CXX_SEEK
+CUDR_CPP = mpicxx -DMPI_GERYON -DUCL_NO_EXIT -DMPICH_IGNORE_CXX_SEEK
 CUDR_OPTS = -O2 $(LMP_INC) # -xHost -no-prec-div -ansi-alias
 
 BIN_DIR = ./
