--- lammps-16Feb16/src/USER-CUDA/verlet_cuda.cpp	2015-10-30 13:04:06.000000000 -0700
+++ patch-files/verlet_cuda.cpp	2016-04-11 10:11:40.793043938 -0700
@@ -63,6 +63,10 @@
 
 VerletCuda::VerletCuda(LAMMPS* lmp, int narg, char** arg) : Verlet(lmp, narg, arg)
 {
+  if (comm->me == 0)
+    error->warning(FLERR,"The USER-CUDA pacakge will be deprecated "
+                   "soon - users should switch to the GPU or KOKKOS packages");
+
   cuda = lmp->cuda;
 
   if(cuda == NULL)
@@ -123,17 +127,6 @@
     fprintf(screen,"  Unit style  : %s\n", update->unit_style);
     fprintf(screen,"  Current step: " BIGINT_FORMAT "\n", update->ntimestep);
     fprintf(screen,"  Time step   : %g\n", update->dt);
-    if (update->max_wall > 0) {
-      char outtime[128];
-      double totalclock = update->max_wall;
-      int seconds = fmod(totalclock,60.0);
-      totalclock  = (totalclock - seconds) / 60.0;
-      int minutes = fmod(totalclock,60.0);
-      int hours = (totalclock - minutes) / 60.0;
-      sprintf(outtime,"  Max walltime: "
-              "%d:%02d:%02d\n", hours, minutes, seconds);
-      fputs(outtime,screen);
-    }
   }
 
   // setup domain, communication and neighboring
@@ -655,11 +648,6 @@
   int firstreneigh = 1;
 
   for(int i = 0; i < n; i++) {
-    if (update->time_expired()) {
-      update->nsteps = i;
-      break;
-    }
-
     if(atom->nlocal == 0)
       error->warning(FLERR, "# CUDA: There are currently no atoms on one of the MPI processes. This is currently prone to encountering errors with USER-CUDA package. Please use the 'processors' keyword to use a more balanced processor layout.");
 
