--- lammps-16Feb16/lib/atc/Makefile.mpic++	2014-12-01 18:12:19.000000000 -0800
+++ patch-files/atc/Makefile.mpic++	2016-04-10 14:35:06.601721739 -0700
@@ -14,8 +14,8 @@
 # include any MPI settings needed for the ATC library to build with
 # must be the same MPI library that LAMMPS is built with
 
-CC =	        mpic++
-CCFLAGS =       -O3 -Wall -g -I../../src -fPIC -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1
+CC =	        mpicxx
+CCFLAGS =       -O2 -Wall -g -I../../src -fPIC -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1 ROLLAVX
 ARCHIVE =	ar
 ARCHFLAG =	-rc
 DEPFLAGS =      -M
