--- lammps-29Oct20/lib/atc/Makefile.mpi	2020-10-29 16:16:31.000000000 -0700
+++ patch-files/atc/Makefile.mpi	2021-05-26 16:37:28.843640507 -0700
@@ -2,7 +2,7 @@
 SHELL = /bin/sh
 
 # which file will be copied to Makefile.lammps
-EXTRAMAKE = Makefile.lammps.linalg
+EXTRAMAKE = Makefile.lammps.empty
 
 # ------ FILES ------
 
@@ -24,7 +24,7 @@
 # must be the same MPI library that LAMMPS is built with
 
 CC =	        mpicxx
-CCFLAGS =       -O3 -Wall -g -fPIC
+CCFLAGS =       -O2 -Wall -fPIC ROLLAVX
 CPPFLAGS = -I../../src -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1
 ARCHIVE =	ar
 ARCHFLAG =	-rc
