--- lammps-17Jan18/lib/atc/Makefile.mpic++	2017-08-09 16:38:53.000000000 -0700
+++ patch-files/atc/Makefile.mpic++	2018-01-29 14:31:50.000000000 -0800
@@ -2,7 +2,7 @@
 SHELL = /bin/sh
 
 # which file will be copied to Makefile.lammps
-EXTRAMAKE = Makefile.lammps.linalg
+EXTRAMAKE = Makefile.lammps.empty
 
 # ------ FILES ------
 
@@ -24,7 +24,7 @@
 # must be the same MPI library that LAMMPS is built with
 
 CC =	        mpicxx
-CCFLAGS =       -O3 -Wall -g -fPIC
+CCFLAGS =       -O2 -Wall -g -fPIC ROLLAVX
 CPPFLAGS = -I../../src -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1
 ARCHIVE =	ar
 ARCHFLAG =	-rc
