--- lammps-22Aug18/src/USER-REAXC/reaxc_defs.h	2017-05-03 08:10:15.000000000 -0700
+++ patch-files/reax/reaxc_defs.h	2018-08-28 17:37:53.829653599 -0700
@@ -95,8 +95,8 @@
 #define MIN_3BODIES    1000
 #define MIN_GCELL_POPL 50
 #define MIN_SEND       100
-#define SAFE_ZONE      1.2
-#define SAFER_ZONE     1.4
+#define SAFE_ZONE      1.6
+#define SAFER_ZONE     1.8
 #define DANGER_ZONE    0.90
 #define LOOSE_ZONE     0.75
 #define MAX_3BODY_PARAM     5
