--- lammps-16Feb16/lib/meam/Makefile.ifort	2013-05-30 12:43:25.000000000 -0700
+++ patch-files/meam/Makefile.ifort	2016-04-10 14:04:40.376946987 -0700
@@ -11,7 +11,7 @@
 
 # which file will be copied to Makefile.lammps
 
-EXTRAMAKE = Makefile.lammps.ifort
+EXTRAMAKE = Makefile.lammps
 
 # ------ FILES ------
 
@@ -26,8 +26,8 @@
 
 # ------ SETTINGS ------
 
-F90 =           ifort
-F90FLAGS =      -O -fPIC
+F90 =           mpif90
+F90FLAGS =      -O -fPIC ROLLAVX
 ARCHIVE =	ar
 ARCHFLAG =	-rc
 USRLIB =
