--- lammps-31Mar17/src/MAKE/OPTIONS/Makefile.kokkos_cuda_openmpi	2016-06-22 09:51:57.000000000 -0700
+++ patch-files/kokkos/Makefile.kokkos_cuda_openmpi	2017-07-11 09:22:07.530043202 -0700
@@ -1,28 +1,29 @@
-# kokkos_cuda = KOKKOS/CUDA package, OpenMPI with nvcc compiler, Kepler GPU
+# kokkos_cuda = KOKKOS package with CUDA support, OpenMPI, nvcc/mpicxx compiler
 
 SHELL = /bin/sh
 
+KOKKOS_ABSOLUTE_PATH = $(shell cd $(KOKKOS_PATH); pwd)
+export MPITYPE = $(KOKKOS_ABSOLUTE_PATH)/config/nvcc_wrapper
+
 # ---------------------------------------------------------------------
 # compiler/linker settings
 # specify flags and libraries needed for your compiler
 
-KOKKOS_ABSOLUTE_PATH = $(shell cd $(KOKKOS_PATH); pwd)
-export OMPI_CXX = $(KOKKOS_ABSOLUTE_PATH)/config/nvcc_wrapper
 CC =		mpicxx
-CCFLAGS =	-g -O3
+CCFLAGS =	-g -std=c++0x
 SHFLAGS =	-fPIC
 DEPFLAGS =	-M
 
 LINK =		mpicxx
-LINKFLAGS =	-g -O3
-LIB = 
+LINKFLAGS =	-g
+LIB = -lstdc++ -LMKLLIB -lmkl_FORTRANTYPE_ilp64 -lmkl_COMPILERTYPE_thread -lmkl_core -lpthread -lgomp GNULIBS -lrt INTELLIB
 SIZE =		size
 
 ARCHIVE =	ar
 ARFLAGS =	-rc
 SHLIBFLAGS =	-shared
 KOKKOS_DEVICES = Cuda, OpenMP
-KOKKOS_ARCH =   Kepler35
+KOKKOS_ARCH = Kepler37:Maxwell50:Maxwell52:Kepler60
 
 # ---------------------------------------------------------------------
 # LAMMPS-specific settings, all OPTIONAL
@@ -115,10 +116,5 @@
 
 # Individual dependencies
 
-depend : fastdep.exe $(SRC)
-	@./fastdep.exe $(EXTRA_INC) -- $^ > .depend || exit 1
-
-fastdep.exe: ../DEPEND/fastdep.c
-	cc -O -o $@ $<
-
-sinclude .depend
+DEPENDS = $(OBJ:.o=.d)
+sinclude $(DEPENDS)
