--- lammps-29Oct20/src/USER-INTEL/pair_tersoff_intel.cpp	2020-10-29 16:16:31.000000000 -0700
+++ patch-files/user-intel/pair_tersoff_intel.cpp	2021-05-26 17:41:25.256842366 -0700
@@ -29,6 +29,10 @@
 #include "memory.h"
 #include "error.h"
 
+#ifdef __INTEL_COMPILER
+#include <mathimf.h>
+#endif
+
 // Currently Intel compiler is required for this pair style.
 // For convenience, base class routines are called if not using Intel compiler.
 #ifndef __INTEL_COMPILER
