--- lammps-23Oct17/src/USER-INTEL/pair_tersoff_intel.cpp	2017-06-20 06:46:34.000000000 -0700
+++ patch-files/user-intel/pair_tersoff_intel.cpp	2018-01-15 16:31:50.464199992 -0800
@@ -28,6 +28,9 @@
 #include "comm.h"
 #include "memory.h"
 #include "error.h"
+#ifdef __INTEL_COMPILER
+#include <mathimf.h>
+#endif
 
 // Currently Intel compiler is required for this pair style.
 // For convenience, base class routines are called if not using Intel compiler.
