--- lammps-16Feb16/lib/awpmd/Makefile.mpicc	2014-09-23 08:55:27.000000000 -0700
+++ patch-files/awpmd/Makefile.mpicc	2016-04-10 15:00:26.637909037 -0700
@@ -36,8 +36,8 @@
 # include any MPI settings needed for the ATC library to build with
 # the same MPI library that LAMMPS is built with
 
-CC =	    mpic++ 
-CCFLAGS = -O -fPIC -Isystems/interact/TCP/ -Isystems/interact -Iivutils/include
+CC =	    mpicxx
+CCFLAGS = -O -fPIC -Isystems/interact/TCP/ -Isystems/interact -Iivutils/include ROLLAVX
 ARCHIVE =	ar
 ARCHFLAG =	-rc
 DEPFLAGS =  -M
