--- lammps-29Oct20/lib/awpmd/Makefile.mpicc	2020-10-29 16:16:31.000000000 -0700
+++ patch-files/awpmd/Makefile.mpicc	2021-05-26 16:41:34.515716772 -0700
@@ -38,7 +38,7 @@
 
 CC =	    mpicxx
 CCFLAGS = -O3 -fPIC -Isystems/interact/TCP/ -Isystems/interact -Iivutils/include \
-	  -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1
+	  -DMPICH_IGNORE_CXX_SEEK -DOMPI_SKIP_MPICXX=1 ROLLAVX
 
 ARCHIVE =	ar
 ARCHFLAG =	-rc
